src/types/custom.rs

```diff
@@ -66,8 +65,13 @@ impl Display for CustomType {
     }
 }
 
-impl From<CustomType> for ClassicType {
-    fn from(ty: CustomType) -> Self {
-        ty.classic_type()
+impl From<CustomType> for SimpleType {
```

Comment:
So this is the API also known as `SimpleType::new_opaque` equivalent to `SimpleType::new_tuple`